common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDBIterator.java

```diff
@@ -272,4 +273,39 @@ static int compare(byte[] a, byte[] b) {
     return a.length - b.length;
   }
 
+  static class ResourceCleaner implements Runnable {
+
+    private final RocksIterator rocksIterator;
+
+    private final AtomicReference<org.rocksdb.RocksDB> _db;
+
+    private final ConcurrentLinkedQueue<Reference<RocksDBIterator<?>>> iteratorTracker;
+
+    public ResourceCleaner(RocksIterator rocksIterator,
+        AtomicReference<org.rocksdb.RocksDB> _db,
```

Comment:
Is it feasible to just pass in `RocksDB` and then retrieve `_db` or `iteratorTracker` when using it?